backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/controller/ElementsController.java

```diff
@@ -0,0 +1,107 @@
+package de.bund.digitalservice.ris.norms.adapter.input.restapi.controller;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import de.bund.digitalservice.ris.norms.adapter.input.restapi.schema.ElementsResponseEntrySchema;
+
+import java.util.List;
+import java.util.Optional;
+
+import de.bund.digitalservice.ris.norms.application.port.input.LoadNormUseCase;
+import de.bund.digitalservice.ris.norms.utils.NodeParser;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+enum ElementType {
+    article,
+}
+
+/**
+ * Controller for norm-related list actions.
+ */
+@RestController
+@RequestMapping(
+        "/api/v1/norms/eli/bund/{agent}/{year}/{naturalIdentifier}/{pointInTime}/{version}/{language}/{subtype}/elements")
+public class ElementsController {
+
+    private final LoadNormUseCase loadNormUseCase;
+
+    public ElementsController(
+            LoadNormUseCase loadNormUseCase
+    ) {
+        this.loadNormUseCase = loadNormUseCase;
+    }
+
+    /**
+     * Retrieves a list of elements inside a norm based on the ELI of the norm and the types of the
+     * elements.
+     *
+     * @param type The type(s) of the elements that should be returned. Elements are returned in the
+     *             order of the types, and then in the order of elements in the norm.
+     * @return A {@link ResponseEntity} containing the list of elements.
+     * <p>Returns HTTP 200 (OK) if the norm is found. The list might be empty.
+     * <p>Returns HTTP 404 (Not Found) if the norm is not found.
+     * <p>Returns HTTP 500 (Server error) if an unsupported type is provided.
+     */
+    @GetMapping(produces = {APPLICATION_JSON_VALUE})
+    public ResponseEntity<List<ElementsResponseEntrySchema>> getList(
+            @PathVariable final String agent,
+            @PathVariable final String year,
+            @PathVariable final String naturalIdentifier,
+            @PathVariable final String pointInTime,
+            @PathVariable final String version,
+            @PathVariable final String language,
+            @PathVariable final String subtype,
+            @RequestParam final ElementType type) {
+
+        // TODO Hannes: Where to put this?
+        // get norm
+        final String eli =
+                buildEli(agent, year, naturalIdentifier, pointInTime, version, language, subtype);
+
+
+        var norm = loadNormUseCase.loadNorm(new LoadNormUseCase.Query(eli));
+
+        if (norm.isEmpty()) return ResponseEntity.notFound().build();
+
+        var articleNodes = NodeParser.getNodesFromExpression("//body//article", norm.get().getDocument());
+
+        var eids = articleNodes.stream().map(node -> NodeParser.getValueFromExpression("./@eId", node)).flatMap(Optional::stream).toList();
+
+        var elements = eids.stream().map(eid ->
+                (ElementsResponseEntrySchema) ElementsResponseEntrySchema.builder()
```

Comment:
@andreasphil Even in this version, I had to type cast the result of the builder. Strange.